R/select_helpers.R

```diff
@@ -1,274 +1,409 @@
 # this function evaluates the select-expression and allows non-standard evaluation
 
 .select_nse <- function(select, data, exclude, ignore_case, regex = FALSE, verbose = FALSE) {
-  # check if data argument is valid
-  if (is.null(data)) {
-    stop(insight::format_message("The `data` argument must be provided."), call. = FALSE)
-  }
 
-  # check data frame input
-  if (!is.null(data) && !is.data.frame(data)) {
-    data <- try(as.data.frame(data), silent = TRUE)
-    if (inherits(data, c("try-error", "simpleError"))) {
-      stop(insight::format_message(
-        "The `data` argument must be a data frame, or an object that can be coerced to a data frame."
-      ), call. = FALSE)
+  .check_data(data)
+  cols <- names(data)
+
+  # avoid conflicts for starts_with(), etc.
+  # (has to be in this order, otherwise error because e.g tidyselect is imported
+  # in tidyr)
+  conflicting_packages <- .conflicting_packages(c("poorman", "tidyr", "dplyr", "tidyselect"))
```

Comment:
I'm not sure we need all packages here, I didn't have any problems with tidyr, dplyr, etc. Or have you found situations where this errors?